### PR TITLE
User management, Movie rating module, fav/disliked/watched 

### PR DESCRIPTION
**Pull request by prymakD**
-User edition(deleting, changing mail/username/bio), email verification before allowing to log in, password recovery, strong password validation
- Movie rating module(set,delete and get requests) and tables to it in DB
- add swagger api documentation available on  [swagger docs]( http://localhost:8080/swagger-ui.html)
- fav/disliked/watched/to_watch features with the ability to get count of how many users added it https://github.com/prymakD/MoviePocket/commit/4f0029eaac7ac3198ecf6308e6814ba2b0e80e2a
- search bar endpoint to search movies by search word(if needed, maybe to be impl on react part)

### CODE REVIEW

File: src/main/java/com/moviePocket/Service/impl/UserServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+    }
+
+    public void setNewUsername(String email, String username) {
+        try {
+            User user = userRepository.findByEmail(email);
+            if (user != null && !userRepository.existsByUsername(username)) {
+                if (username.isEmpty()) {
+                    throw new MessagingException("Username cannot be empty");
+                }
+                user.setUsername(username);
```
Comment (AntonPazniak) [inline src/main/java/com/moviePocket/Service/impl/UserServiceImpl.java:148]: you removed the check for whether the username is busy

**Review by AntonPazniak — APPROVED** [review]
code checked

File: src/main/java/com/moviePocket/Service/impl/UserServiceImpl.java (hunk 1 above)
Comment (prymakD) [reply src/main/java/com/moviePocket/Service/impl/UserServiceImpl.java:148]: yeah, you right, my bad

**Review by EkintoQ — APPROVED** [review]
Code is fine and generally clean.
Need more tests for testing app on backend side.

**Pull request merged**